Realm/RealmSwift/Realm.swift

```diff
@@ -58,15 +58,31 @@ public class Realm {
     }
 
     public convenience init(path: String) {
-        self.init(path: path, readOnly: false, error: nil)
+        self.init(rlmRealm: RLMRealm(path: path, readOnly: false, error: nil))
     }
 
     public convenience init(inMemoryIdentifier: String) {
         self.init(rlmRealm: RLMRealm.inMemoryRealmWithIdentifier(inMemoryIdentifier))
     }
 
-    public convenience init(path: String, readOnly readonly: Bool, error: NSErrorPointer) {
-        self.init(rlmRealm: RLMRealm(path: path, readOnly: readonly, error: error))
+    public convenience init?(path: String, readOnly readonly: Bool, error: NSErrorPointer) {
+        if let rlmRealm = RLMRealm(path: path, readOnly: readonly, error: error) as RLMRealm? {
+            self.init(rlmRealm: rlmRealm)
+        }
+        else {
+            self.init(rlmRealm: RLMRealm.inMemoryRealmWithIdentifier(NSUUID().UUIDString))
```

Comment:
We should create a no-op function that "returns" a nil RLMRealm in Objective-C and call that instead of creating an in-memory realm. The interface for this can be in `Realm.Private`.